docs/DeveloperGuide.md

```diff
@@ -1,38 +1,180 @@
 # Developer Guide
 
 ## Acknowledgements
-
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
+
+### Architecture
+The ***Architecture Diagram*** given above explains the high-level design of the App.
+Given below is a quick overview of main components and how they interact with each other.
+
+<img src="images/ArchitectureDiagram.png" width="300" />
+
+**Main components of the architecture**
+
+## **Implementation**
+
+### Add transaction feature
+The add transaction feature is facilitated by the `Parser` class which parses user input and creates a `Expense` or
+`Allowance` object which extends from the `Transaction` class. The created `Transaction` object will be stored in a 
+`TransactionList`.
+
+Given below is an example usage scenario and how the add transaction mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `TransactionList` will be initialized.
+
+Step 2. The user executes `expense /amt 20 /date 24-10-2023 /desc Lunch /note Pasta /cat Food` command to create a
+transaction. The `expense` command calls `Parser#parseExpense()` to create an `Expense` object. The 
+`AddCommand#execute()` is then called to store the `Expense` object in the `TransactionList`.
+
+The following sequence diagram shows how the add transaction operation works:
+
+<img src="images/AddTransactionSequenceDiagram.png" width="600" />
```

Comment:
<img width="514" alt="image" src="https://github.com/nus-cs2113-AY2324S1/tp/assets/84222810/31999f8b-766d-4873-ae4e-a20112375ab2">

Maybe you can omit the classes and User at the bottom because it might not be necessary to repeat them again? Same for the other sequence diagrams